### PR TITLE
Lofreq improvements

### PR DESCRIPTION
**Pull request by rdmorin**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [ ] I used the cookiecutter template and updated the placeholder rules.

- [ ] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [ ] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [ ] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [ ] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [ ] I updated the final target rule (`*_all`) to include every output rule.

- [ ] I explained important module design decisions in `CHANGELOG.md`.

- [ ] I tested the module on real data for all supported `seq_type` values.

- [ ] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [ ] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [ ] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [ ] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [ ] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [ ] I added more granular output subdirectories.

- [ ] I added rules to the `reference_files` workflow to generate any new reference files.

- [ ] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [ ] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

To be completed.

### CODE REVIEW

File: modules/lofreq/1.0/src/lofreq2_call_pparallel.py [hunk 1]
```diff
@@ -459,7 +459,7 @@ def main():
     if idx != -1:
         final_vcf_out = lofreq_call_args[idx+1]
         if os.path.exists(final_vcf_out):
-            LOG.fatal("Cowardly refusing to overwrite already existing"
+            LOG.fatal("On line 462 Cowardly refusing to overwrite already existing"
```
Comment (Kdreval) [inline modules/lofreq/1.0/src/lofreq2_call_pparallel.py]: very nice!!

File: modules/sequenza/1.4/src/samtools.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import shlex
```
Comment (Kdreval) [inline modules/sequenza/1.4/src/samtools.py]: Where is this script used? I don't think it is referenced in the Lofreq.smk or module config

Comment (rdmorin) [reply modules/sequenza/1.4/src/samtools.py]: Oops. This shouldn't be here.

Comment (rdmorin) [reply modules/sequenza/1.4/src/samtools.py]: I'll remove for final PR once I move everything to the new version directory

File: modules/lofreq/1.0/lofreq.smk [hunk 3]
```diff
@@ -101,22 +113,76 @@ rule _lofreq_run:
...
+        opts = CFG["options"]["lofreq"]
+    conda:
+        CFG["conda_envs"]["lofreq"]
+    threads:
+        CFG["threads"]["lofreq"]
+    resources:
+        **CFG["resources"]["lofreq"]
+    shell:
+        op.as_one_line("""
+        ln -f {input.vcf_relaxed} {output.out_dir};
```
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/lofreq/1.0/lofreq.smk]: On purpose, yes. It effectively makes a redundant copy of the file so other parallel processes can use it and it only gets completely deleted once the last job is done.

File: modules/lofreq/1.0/lofreq.smk [hunk 4]
```diff
@@ -101,22 +113,76 @@ rule _lofreq_run:
...
-            &&
-            lofreq somatic {params.opts} --threads {threads} -t {input.tumour_bam} -n {input.normal_bam}
-            -f {input.fasta} -o $(dirname {output.vcf_snvs_filtered})/ -d {input.dbsnp} {params.regions} --bed {input.bed}
-            > {log.stdout} 2> {log.stderr}
-            &&
-            rm {params.rm_files};
+            touch {output.preprocessing_start}
+            && 
+            lofreq somatic --normal_only {params.opts} --threads {threads} -t {input.normal_bam} -n {input.normal_bam}
+            -f {input.fasta} -o {output.out_dir}/ -d {input.dbsnp} --bed {input.bed}
```
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/lofreq/1.0/lofreq.smk]: Doesn't really matter if it does. The directory won't actually have that second slash.

**Review by Kdreval — DISMISSED** [review]

**Review by Kdreval — APPROVED** [review]

**Pull request merged**